Source/WebCore/rendering/svg/RenderSVGResourceMasker.cpp

```diff
@@ -89,31 +99,32 @@ bool RenderSVGResourceMasker::applyResource(RenderElement& renderer, const Rende
         }
 #endif
         // FIXME (149470): This image buffer should not be unconditionally unaccelerated. Making it match the context breaks alpha masking, though.
-        maskerData->maskImage = context->createScaledImageBuffer(repaintRect, scale, maskColorSpace, RenderingMode::Unaccelerated);
-        if (!maskerData->maskImage)
+        maskerData.maskImage = context->createScaledImageBuffer(maskerData.inputs.repaintRect, maskerData.inputs.scale, maskColorSpace, RenderingMode::Unaccelerated);
+        if (!maskerData.maskImage)
             return false;
 
-        if (!drawContentIntoMaskImage(maskerData, drawColorSpace, &renderer))
-            maskerData->maskImage = nullptr;
+        if (!drawContentIntoMaskImage(maskerData, drawColorSpace))
+            maskerData.maskImage = nullptr;
+
+        createdNewMaskImage = true;
     }
 
-    if (!maskerData->maskImage)
+    if (!maskerData.maskImage)
         return false;
 
-    SVGRenderingContext::clipToImageBuffer(*context, repaintRect, scale, maskerData->maskImage, missingMaskerData);
+    SVGRenderingContext::clipToImageBuffer(*context, maskerData.inputs.repaintRect, maskerData.inputs.scale, maskerData.maskImage, createdNewMaskImage);
     return true;
 }
 
-bool RenderSVGResourceMasker::drawContentIntoMaskImage(MaskerData* maskerData, const DestinationColorSpace& colorSpace, RenderObject* object)
+bool RenderSVGResourceMasker::drawContentIntoMaskImage(MaskerData& maskerData, const DestinationColorSpace& colorSpace)
 {
-    GraphicsContext& maskImageContext = maskerData->maskImage->context();
+    GraphicsContext& maskImageContext = maskerData.maskImage->context();
 
     // Eventually adjust the mask image context according to the target objectBoundingBox.
     AffineTransform maskContentTransformation;
-    if (maskElement().maskContentUnits() == SVGUnitTypes::SVG_UNIT_TYPE_OBJECTBOUNDINGBOX) {
-        FloatRect objectBoundingBox = object->objectBoundingBox();
-        maskContentTransformation.translate(objectBoundingBox.location());
-        maskContentTransformation.scale(objectBoundingBox.size());
+    if (maskerData.inputs.objectBoundingBox) {
+        maskContentTransformation.translate(maskerData.inputs.objectBoundingBox->location());
+        maskContentTransformation.scale(maskerData.inputs.objectBoundingBox->size());
```

Comment:
This can be simplified 

```
if (auto& maskContentTransformation = maskerData.inputs.maskContentTransformation)
    maskImageContext.concatCTM(*maskContentTransformation);
```